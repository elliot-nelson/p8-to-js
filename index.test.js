const p8tojs = require('.');

const PICO8_CARTRIDGE_1 = `pico-8 cartridge // http://www.pico-8.com
version 32
__lua__

__gfx__
00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
__sfx__
011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000
011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
__music__
00 00424344
`;

describe('p8tojs', () => {
    describe('convert', () => {
        it('creates a JS file with a default export', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1)
            ).toMatchSnapshot();
        });

        it('for export=default, creates a JS file with a default export', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1, { export: 'default' })
            ).toMatchSnapshot();
        });

        it('for export=commonjs, creates a JS file with a module.exports', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1, { export: 'commonjs' })
            ).toMatchSnapshot();
        });

        it('for export=json, creates a JSON file', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1, { export: 'json' })
            ).toMatchSnapshot();
        });

        it('creates a JS file with a custom export variable', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1, { export: 'Cartridge' })
            ).toMatchSnapshot();
        });

        it('limits the selected sections if sections are specified', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1, { sections: ['lua', 'sfx'] })
            ).toMatchSnapshot();
        });

        it('exports with base64 encoding if specified', () => {
            expect(
                p8tojs.convert(PICO8_CARTRIDGE_1, { encoding: 'base64' })
            ).toMatchSnapshot();
        });
    });
});
