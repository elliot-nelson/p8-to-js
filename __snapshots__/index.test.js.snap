// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p8tojs convert creates a JS file with a custom export variable 1`] = `"export const Cartridge = { gfx: '00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000', lua: '', music: '00 00424344', sfx: '011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000\\\\n011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000' };"`;

exports[`p8tojs convert creates a JS file with a default export 1`] = `"export default { gfx: '00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000', lua: '', music: '00 00424344', sfx: '011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000\\\\n011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000' };"`;

exports[`p8tojs convert exports with base64 encoding if specified 1`] = `"export default { gfx: 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==', lua: '', music: 'AA==', sfx: 'ARQAAA8FAPBQDAUA8FAMBQDwUAwFAPBQDAUA8FAMBQDwUAwFAPBQDwUAwFAPBQDAUA8FAPBQDAUA8FAMBQDwUAwFAPBQDAUA8FAPBQDwUA8FAAAA' };"`;

exports[`p8tojs convert for export=commonjs, creates a JS file with a module.exports 1`] = `"module.exports = { gfx: '00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000', lua: '', music: '00 00424344', sfx: '011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000\\\\n011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000' };"`;

exports[`p8tojs convert for export=default, creates a JS file with a default export 1`] = `"export default { gfx: '00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000', lua: '', music: '00 00424344', sfx: '011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000\\\\n011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000' };"`;

exports[`p8tojs convert for export=json, creates a JSON file 1`] = `
"{
  \\"lua\\": \\"\\",
  \\"gfx\\": \\"00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00077000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\\\n00700700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\",
  \\"sfx\\": \\"011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000\\\\n011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\\",
  \\"music\\": \\"00 00424344\\"
}"
`;

exports[`p8tojs convert limits the selected sections if sections are specified 1`] = `"export default { lua: '', sfx: '011400000f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500f0500c0500f0500c0500f0500c0500f0500c0500f0500f0500f0500f05000000\\\\n011000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000' };"`;
